### PR TITLE
Added support for user bound SUMMA depth

### PR DESCRIPTION
**Pull request by justusc**
- Added support to SUMMA for memory bound depth via the `TA__SUMMA__MAX_MEMORY` environment variables
- Added support to SUMMA for hard depth bound via `TA__SUMMA__MAX_DEPTH` environment variables

### CODE REVIEW

File: src/TiledArray/dist_eval/contraction_eval.h [hunk 1]
```diff
@@ -104,6 +107,46 @@ namespace TiledArray {
...
+        if(max_memory) {
+            // Convert the string into bytes
+            std::stringstream ss(max_memory);
+            double memory = 0.0;
+            if(ss >> memory) {
+                if(memory > 0.0) {
+                    std::string unit;
+                    if(ss >> unit) { // Failure == assume bytes
+                        if(unit == "KB" || unit == "kB") {
+                            memory *= 1024.0;
```
Comment (evaleev) [inline src/TiledArray/dist_eval/contraction_eval.h]: let's use SI system(https://en.wikipedia.org/wiki/Binary_prefix): change 1024 -> 1e3, etc. Feel free to add "KiB", "MiB", "GiB" units

File: src/TiledArray/dist_eval/contraction_eval.h [hunk 2]
```diff
@@ -104,6 +107,46 @@ namespace TiledArray {
 
     private:
 
+      // Static variable initialization ----------------------------------------
+
+
+      /// Initialize max_memory_ limit for SUMMA
+      static size_type init_max_memory() {
+        const char* max_memory = getenv("TA__SUMMA__MAX_MEMORY");
```
Comment (evaleev) [inline src/TiledArray/dist_eval/contraction_eval.h]: Q: why double underscore around SUMMA?

File: src/TiledArray/dist_eval/contraction_eval.h (hunk 1 above)
Comment (justusc) [inline src/TiledArray/dist_eval/contraction_eval.h]: That seems reasonable. I will have some time to twerk and test this weekend.

**Comment by justusc** [comment]
FYI, the memory limit is a soft limit, based on the assumption that the tile sizes do not vary too much.

**Pull request merged**